File: GDTMobApp/GDTAds/GDTUnifiedNativeAd/Cell/UnifiedNativeAdThreeImageCell.m
//
//  UnifiedNativeAdThreeImageCell.m
//  GDTMobApp
//
//  Created by royqpwang on 2019/5/19.
//  Copyright © 2019 Tencent. All rights reserved.
//

#import "UnifiedNativeAdThreeImageCell.h"
#import "UnifiedNativeAdCustomView.h"

@interface UnifiedNativeAdThreeImageCell()

@end

@implementation UnifiedNativeAdThreeImageCell


#pragma mark - public
- (void)setupWithUnifiedNativeAdDataObject:(GDTUnifiedNativeAdDataObject *)dataObject delegate:(id<GDTUnifiedNativeAdViewDelegate>)delegate vc:(UIViewController *)vc
{
    CGFloat imageRate = 228 / 150.0; // 三小图默认比例
    if (dataObject.imageHeight > 0) {
        imageRate = dataObject.imageWidth / (CGFloat)dataObject.imageHeight;
    }
    CGFloat width = [UIScreen mainScreen].bounds.size.width - 16;
    self.adView.backgroundColor = [UIColor grayColor];
    self.adView.titleLabel.frame = CGRectMake(8, 8, 250, 30);
    self.adView.descLabel.frame = CGRectMake(8, 46, width, 30);
    CGFloat imageWidth = (width - 16) / 3;
    self.adView.leftImageView.frame = CGRectMake(8, 84, imageWidth, imageWidth / imageRate);
    self.adView.midImageView.frame = CGRectMake(16 + imageWidth, 84, imageWidth, imageWidth / imageRate);
    self.adView.rightImageView.frame = CGRectMake(24 + imageWidth * 2, 84, imageWidth, imageWidth / imageRate);
    self.adView.frame = CGRectMake(0, 0, [UIScreen mainScreen].bounds.size.width, 92 + imageWidth / imageRate);
    self.adView.logoView.frame = CGRectMake(CGRectGetWidth(self.adView.frame) - kGDTLogoImageViewDefaultWidth, CGRectGetHeight(self.adView.frame) - kGDTLogoImageViewDefaultHeight, kGDTLogoImageViewDefaultWidth, kGDTLogoImageViewDefaultHeight);
    self.adView.viewController = vc;
    self.adView.delegate = delegate;
    [self.adView setupWithUnifiedNativeAdObject:dataObject];
    [self.adView.clickButton sizeToFit];
    self.adView.clickButton.frame = (CGRect) {
        .origin.x = width - 8 - self.adView.clickButton.frame.size.width,
        .origin.y = 8,
        .size = self.adView.clickButton.frame.size
    };
    if ([dataObject isAdValid]) {
        [self.adView registerDataObject:dataObject clickableViews:@[self.adView.clickButton,
                                                                    self.adView.leftImageView,
                                                                    self.adView.midImageView,
                                                                    self.adView.rightImageView]];
    }
}

+ (CGFloat)cellHeightWithUnifiedNativeAdDataObject:(GDTUnifiedNativeAdDataObject *)dataObject
{
    CGFloat height = 0;
    CGFloat imageRate = 228 / 150.0; // 三小图默认比例
    if (dataObject.imageHeight > 0) {
        imageRate = dataObject.imageWidth / (CGFloat)dataObject.imageHeight;
    }
    CGFloat width = [UIScreen mainScreen].bounds.size.width - 16;
    CGFloat imageWidth = (width - 16) / 3;
    height = 92 + imageWidth / imageRate;
    return height;
}

@end

File: GDTMobSample-Swift/GDTAdTypes/UnifiedNativeAdViewController/cell/UnifiedNativeAdThreeImageCell.swift
//
//  UnifiedNativeAdThreeImageCell.swift
//  GDTMobSample-Swift
//
//  Created by 胡城阳 on 2019/10/28.
//  Copyright © 2019 Tencent. All rights reserved.
//

import UIKit

class UnifiedNativeAdThreeImageCell: UnifiedNativeAdBaseTableViewCell {

    override func setupWithUnifiedNativeAdDataObject(dataObject: GDTUnifiedNativeAdDataObject, delegate: GDTUnifiedNativeAdViewDelegate, vc: UIViewController) {
        
        var imageRate:CGFloat = 228 / 150.0
        if dataObject.imageHeight > 0 {
            imageRate = CGFloat(dataObject.imageWidth / dataObject.imageHeight)
        }
        let width = UIScreen.main.bounds.width - 16
        self.adView.backgroundColor = UIColor.gray
        self.adView.clickButton.frame = CGRect(x: width - 68, y: 8, width: 60, height: 44)
        self.adView.titleLabel.frame = CGRect(x: 8, y: 8, width: 250, height: 30)
        self.adView.descLabel.frame = CGRect(x: 8, y: 46, width: width, height: 30)
        let imageWidth:CGFloat = (width - 16) / 3
        self.adView.leftImageView.frame = CGRect(x: 8, y: 84, width: imageWidth, height: imageWidth / imageRate)
        self.adView.midImageView.frame = CGRect(x: 16 + imageWidth, y: 84, width: imageWidth, height: imageWidth / imageRate)
        self.adView.rightImageView.frame = CGRect(x: 24 + imageWidth * 2, y: 84, width: imageWidth, height: imageWidth / imageRate)
        self.adView.frame = CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 92 + imageWidth / imageRate)
        self.adView.logoView.frame = CGRect(x: self.adView.frame.width - kGDTLogoImageViewDefaultWidth, y: self.adView.frame.height - kGDTLogoImageViewDefaultHeight, width: kGDTLogoImageViewDefaultWidth, height: kGDTLogoImageViewDefaultHeight)
        self.adView.viewController = vc
        self.adView.delegate = delegate
        self.adView.setupWithUnifiedNativeAdObject(unifiedNativeDataObject: dataObject)
        self.adView.registerDataObject(dataObject, clickableViews: [self.adView.clickButton,self.adView.leftImageView,self.adView.midImageView,self.adView.rightImageView])
        
    }
    
    override class func cellHeightWithUnifiedNativeAdDataObject(dataObject:GDTUnifiedNativeAdDataObject)->CGFloat{
        var height:CGFloat = 0
        var imageRate:CGFloat = 228 / 150.0
        if dataObject.imageHeight > 0 {
            imageRate = CGFloat(dataObject.imageWidth / dataObject.imageHeight)
        }
        let width:CGFloat = UIScreen.main.bounds.width - 16
        let imageWidth = (width - 16) / 3;
        height = 92 + imageWidth / imageRate
        return height
    }

}
